### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Msawekho**
I created a to-do list that adds, remove/delete, search and mark completed tasks .

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -10,9 +10,14 @@
...
+            <div class="add-task">
+                <input type="text" id="new-task" placeholder="Add a new task..." />
+                <button onclick="addTask()">Add Task</button>
+            </div>
+            <div class="task-list-section">
+                <input type="text" id="search-task" placeholder="Search tasks..." oninput="searchTask()" />
+                <ul id="task-list"></ul>
+            </div>
         </div>
     </div>
```
Comment (MulamuMammba) [inline public/index.html]: # Use the node backend
@Msawekho from what I can see you are not utilizing the backend node server and doing everything on the fronend

**Review by MulamuMammba — CHANGES_REQUESTED** [review]
@Msawekho Add the necessary connection between the frontend and backend

File: server.js [hunk 2]
```diff
@@ -1,15 +1,57 @@
...
+
 const port = 3000;
 
-app.use(express.static(path.join(__dirname, 'public'))); // Serve static files
+app.set("view engine", "js");
+app.use(express.urlencoded({ extended: true }));
+app.use(express.static('public'));
+
+let tasks=[];
+
```
Comment (Msawekho) [inline server.js]: @Ntlahla80   is the linking correct for the node server?

File: server.js [hunk 3]
```diff
@@ -1,15 +1,57 @@
...
 const app = express();
+
 const port = 3000;
 
-app.use(express.static(path.join(__dirname, 'public'))); // Serve static files
+app.set("view engine", "js");
+app.use(express.urlencoded({ extended: true }));
+app.use(express.static('public'));
+
+let tasks=[];
```
Comment (Ntlahla80) [inline server.js]: @Msawekho I see you have updated the node server.js nice job!!

File: public/style.css [hunk 4]
```diff
@@ -12,15 +11,30 @@ body {
...
     padding: 20px;
     border-radius: 10px;
     box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
-    max-width: 400px;
+    max-width: 600px;
     width: 100%;
+    display: flex;
+    flex-direction: column;
+    gap: 10px;
 }
```
Comment (Ntlahla80) [inline public/style.css]: @Msawekho I like how you have styled your front end nice job!!

**Review by Ntlahla80 — APPROVED** [review]

**Review by MulamuMammba — APPROVED** [review]
I checked the improvements, looks good to me.

**Review by ClementRep — APPROVED** [review]

**Pull request merged**